src/server.rs

```diff
@@ -15,113 +14,150 @@ use crate::configuration::Configuration;
 use crate::kvstore::KvStore;
 use crate::logger::{Logger, LogLevel};
 
+// TODO: passing configuration to Server
 pub struct Server {
     endpoint: String,
-    use_tls: bool
+    use_tls: bool,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
 struct ErrorMessage {
-    message: String,
+    code: i32,
+    message: &'static str,
     details: Option<String>
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+struct Claims {
+    sub: String,
+    iss: String,
+    iat: i64,
+    exp: i64,
+}
+
 // TODO: move into implementation
-fn handler_vuejs<'a>(_: &mut Request, res: Response<'a>) -> MiddlewareResult<'a> {
+fn middleware_webui<'a>(_: &mut Request, res: Response<'a>) -> MiddlewareResult<'a> {
     let mut data = HashMap::<&str, &str>::new();
     data.insert("name", "Alex");
-    res.render("webui/dist/index.tpl", &data)
+    res.render("webui/dist/index.html", &data)
```

Comment:
Remove the data injection.

Add the following URI query parameters:
 - `endpoint`
 - `token`

This will be used to auto-connect to the Web UI.

Reply:
For me, it's better to have an injection as a "bootstrap" process and not a query parameter that could be used as a security breach, but we need to think about.

Reply:
I can't read the injected data. Passing it as query parameters is not a security problem...

Reply:
All it's possible, it's the life

Reply:
You can, it's absolutely not a problem, but we will see